src/BlockFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusCMSPlugin;
+
+use Setono\SyliusCMSPlugin\Doctrine\ORM\BlockRepository;
+use Setono\SyliusCMSPlugin\Model\Block;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class BlockFactory implements FactoryInterface
+{
+    private BlockRepository $repository;
+
+    public function __construct(
+        BlockRepository $repository
+    ) {
+        $this->repository = $repository;
+    }
+
+    public function createNew(): Block
+    {
+        return new Block();
+    }
+
+    public function createNewFromResource(string $resourceId): Block
```

Comment:
```suggestion
    public function createNewFromResource(string $resourceId): BlockInterface
```

The id should be an int, shouldn't it?

Reply:
Yeah you are right, I should update the route to have `requirements` and make it an int